compiler/include/genret.h

```diff
@@ -96,10 +96,14 @@ class GenRet {
   // one of the following is set when generating LLVM
   llvm::Value *val; // use val->getType() to obtain LLVM type
   llvm::Type *type; // set when generating a type only
+  Type *baseType; // surrounding structure, if this is a field
```

Comment:
We might be able to choose a better field name than `baseType` but it is OK and I appreciate the comment.